libs/datamodel/core/src/transform/ast_to_dml/db.rs

```diff
@@ -51,28 +47,19 @@ use names::Names;
 /// lifetime management simple.
 pub(crate) struct ParserDatabase<'ast> {
     ast: &'ast ast::SchemaAst,
-    datasource: Option<&'ast Datasource>,
     names: Names<'ast>,
     types: Types<'ast>,
     relations: Relations<'ast>,
-    pub(crate) preview_features: BitFlags<PreviewFeature>,
 }
 
 impl<'ast> ParserDatabase<'ast> {
     /// See the docs on [ParserDatabase](/struct.ParserDatabase.html).
-    pub(super) fn new(
-        ast: &'ast ast::SchemaAst,
-        datasource: Option<&'ast Datasource>,
-        diagnostics: Diagnostics,
-        preview_features: BitFlags<PreviewFeature>,
-    ) -> (Self, Diagnostics) {
+    pub(super) fn new(ast: &'ast ast::SchemaAst, diagnostics: Diagnostics) -> (Self, Diagnostics) {
```

Comment:
Does the removal of the preview flags mean we're now always going to parse new features and have an additional validation step later checking for enabled preview flags?

Reply:
Yep, that's how it works. In practice the validations in `validations.rs` are nicer, so I'm sure it will work out. It's not _strictly necessary_ (we could have separate crates for datasource and preview features that parser database would then depend on), but I think it was the most straightforward for now. There are ways out of this design decision if it doesn't work out.